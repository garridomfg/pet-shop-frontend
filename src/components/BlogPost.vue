<template>
  <div class="d-flex">
    <v-responsive>
      <v-img :height="400" cover :src="getImageUrl(props.blog.metadata.image)"></v-img>
    </v-responsive>
    <div class="pa-16 w-50">
      <p class="blog-title mb-4">{{ props.blog.title }}</p>
      <v-btn color="primary" variant="flat">
        <span class="text-white">
          {{ props.blog.title }}
        </span>
      </v-btn>
    </div>
  </div>
</template>
<script setup lang="ts">
import useProducts from '../composables/useProducts'
import { Data } from '../interfaces/products'

const props = defineProps<{
  blog: Data
}>()

const { getImageUrl } = useProducts()
</script>
