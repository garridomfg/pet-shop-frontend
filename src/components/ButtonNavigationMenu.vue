<template>
  <div class="d-flex align-center">
    <v-overlay class="d-flex align-center justify-center" @click:outside="isLogin = true">
      <template #activator="{ props }">
        <v-btn
          v-bind="button.label === 'LOGIN' ? props : null"
          class="mx-3 text-white"
          size="large"
          variant="outlined"
          v-for="button in props.buttons"
          :key="button.label"
          @click="button.fn"
          :prepend-icon="button.icon"
        >
          <template v-if="button.icon ? button.icon : undefined" v-slot:prepend>
            <v-icon color="success"></v-icon>
          </template>
          {{ button.label }}
          <span v-if="button.count >= 0">({{ button.count }})</span>
        </v-btn>
      </template>
      <login-form v-if="isLogin" :toggle-login-register="toggleLoginRegister" />
      <register-form v-else :toggle-login-register="toggleLoginRegister" />
    </v-overlay>
  </div>
</template>
<script setup lang="ts">
import useLogin from '../composables/useLogin'
import { NavButtons } from '../interfaces/navbar'
import { LoginForm, RegisterForm } from '.'

const emit = defineEmits<{
  (e: 'on:toggle-menu', isLogin: boolean): void
}>()

const props = defineProps<{
  buttons: NavButtons[]
}>()

const { isLogin, toggleLoginRegister } = useLogin(emit)
</script>
