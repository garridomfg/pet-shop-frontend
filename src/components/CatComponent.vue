<template>
  <div>
    Cat
  </div>
</template>
<script setup lang="ts">
</script>
<style>
  
</style>