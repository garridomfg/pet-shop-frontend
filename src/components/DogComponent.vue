<template>
  <div>
    Dog
  </div>
</template>
<script setup lang="ts">
</script>
<style>
  
</style>