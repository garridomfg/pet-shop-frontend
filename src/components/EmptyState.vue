<template>
  <v-sheet
    class="d-flex align-center justify-center flex-wrap text-center mx-auto px-4"
    elevation="2"
    height="250"
    rounded
    max-width="800"
    width="100%"
  >
    <div>
      <h2 class="text-h4 font-weight-black" style="color: var(--primary500)">We are sorry</h2>

      <div class="text-h5 font-weight-medium mb-2">There is nothing to show here</div>
    </div>
  </v-sheet>
</template>
<script setup lang="ts"></script>
