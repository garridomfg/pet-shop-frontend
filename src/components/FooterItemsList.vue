<template>
  <div class="ml-16 d-flex flex-column" v-for="item in footerItems" :key="item.title">
    <h4 class="text-black font-weight-bold">{{ item.title }}</h4>
    <a v-for="option in item.options" :key="option.label" :href="option.path" class="mt-1">
      {{ option.label }}
    </a>
  </div>
</template>
<script setup lang="ts">
import useFooter from '../composables/useFooter'
const { footerItems } = useFooter()
</script>
