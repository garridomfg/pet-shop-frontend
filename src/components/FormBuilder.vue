<template>
  <div class="pa-2" v-for="schema in fields" :key="schema.name" :class="schema.class">
    <Field
      v-model="schema.value"
      :type="schema.type"
      :name="schema.name"
      :rules="schema.rules"
      :as="schema.as || ''"
      v-slot="{ field, meta, errorMessage }"
    >
      <v-text-field
        v-bind="field"
        v-if="schema.type !== 'checkbox'"
        :label="schema.label"
        :type="schema.type"
        variant="outlined"
        :error="(meta.touched && !meta.valid) || !!errorMessage"
        :error-messages="errorMessage"
        :readonly="schema.readonly"
      ></v-text-field>
      <v-checkbox v-else v-model="schema.value" :label="schema.label"></v-checkbox>
      <ErrorMessage class="text-red" :name="schema.name" />
    </Field>
  </div>
</template>
<script setup lang="ts">
import { Field, ErrorMessage } from 'vee-validate'
import { SchemaField } from '../interfaces/login'

defineProps<{
  fields: SchemaField[]
}>()
</script>
