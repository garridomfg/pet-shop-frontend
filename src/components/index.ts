import BlogPost from './BlogPost.vue'
import ButtonNavigationMenu from './ButtonNavigationMenu.vue'
import FooterItemsList from './FooterItemsList.vue'
import NavBar from './NavBar.vue'
import NavigationMenu from './NavigationMenu.vue'
import ProductSlider from './ProductSlider.vue'
import PromotionsCarousel from './PromotionsCarousel.vue'
import SearchInput from './SearchInput.vue'
import SearchInputList from './SearchInputList.vue'
import ShopFooter from './ShopFooter.vue'

export {
  BlogPost,
  ButtonNavigationMenu,
  FooterItemsList,
  NavBar,
  NavigationMenu,
  ProductSlider,
  PromotionsCarousel,
  SearchInput,
  SearchInputList,
  ShopFooter,
}
