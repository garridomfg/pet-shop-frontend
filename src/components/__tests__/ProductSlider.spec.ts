import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { createPinia, setActivePinia } from 'pinia'
import { createRouter, createWebHistory } from 'vue-router'

import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

import { routes } from '../../router'
import { useUsersStore } from '../../stores/users'
import { ProductSlider } from '..'
import { products } from './products.mock'

const vuetify = createVuetify({
  components,
  directives
})

const router = createRouter({
  history: createWebHistory(),
  routes
})

global.ResizeObserver = require('resize-observer-polyfill')

describe('Product Slider Tests', () => {
  let wrapper
  let usersStore
  const mockRouter = {
    push: vi.fn()
  }
  beforeEach(() => {
    setActivePinia(createPinia())
    usersStore = useUsersStore()

    wrapper = mount(ProductSlider, {
      props: {
        title: 'Pet Clean-Up And Odor Control',
        categoryId: '4bd58125-b298-3dc4-97f6-fc275d106c2a',
        products
      },
      global: {
        plugins: [vuetify, router],
        mocks: {
          $router: mockRouter
        }
      }
    })
  })
  afterEach(() => {
    vi.restoreAllMocks()
  })

  test('should render the component correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render the title prop correctly', () => {
    expect(wrapper.find('p').text()).toBe('Pet Clean-Up And Odor Control')
  })
})
