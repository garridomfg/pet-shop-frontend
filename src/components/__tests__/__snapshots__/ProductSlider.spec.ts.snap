// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product Slider Tests > should render the component correctly 1`] = `
"<div>
  <p class=\\"slider-title text-capitalize\\">Pet Clean-Up And Odor Control</p>
  <div class=\\"v-slide-group pa-4\\" tabindex=\\"0\\">
    <!---->
    <div class=\\"v-slide-group__container\\">
      <div class=\\"v-slide-group__content\\" style=\\"transform: translateX(0px);\\">
        <div class=\\"slider-card\\">
          <div class=\\"v-responsive\\">
            <div class=\\"v-responsive__sizer\\"></div>
            <!---->
            <div class=\\"v-responsive__content\\">
              <div class=\\"v-responsive v-img v-img--booting\\" style=\\"height: 220px;\\">
                <div class=\\"v-responsive__sizer\\"></div>
                <transition-stub name=\\"fade-transition\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><img class=\\"v-img__img v-img__img--cover\\" src=\\"http://pet-shop.buckhill.com.hr/api/v1/file/46b68633-a0f2-3b4e-8c0e-470e04b7362d\\" style=\\"display: none;\\"></transition-stub>
                <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                  <!---->
                </transition-stub>
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <p class=\\"text-decoration-underline mt-4 mb-2 font-weight-bold text-h6\\">Shout for Pets Odor and Urine Eliminator - Effective Way to Remove Puppy &amp; Dog Odors and Stains from Carpets &amp; Rugs</p>
          <p class=\\"text-capitalize\\">pet clean-up and odor control</p>
          <p class=\\"font-weight-bold text-h6\\">15.1</p>
        </div>
        <div class=\\"slider-card\\">
          <div class=\\"v-responsive\\">
            <div class=\\"v-responsive__sizer\\"></div>
            <!---->
            <div class=\\"v-responsive__content\\">
              <div class=\\"v-responsive v-img v-img--booting\\" style=\\"height: 220px;\\">
                <div class=\\"v-responsive__sizer\\"></div>
                <transition-stub name=\\"fade-transition\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><img class=\\"v-img__img v-img__img--cover\\" src=\\"http://pet-shop.buckhill.com.hr/api/v1/file/5cf8b142-c5d8-3ad4-bde9-bb202adeadd1\\" style=\\"display: none;\\"></transition-stub>
                <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                  <!---->
                </transition-stub>
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <p class=\\"text-decoration-underline mt-4 mb-2 font-weight-bold text-h6\\">Amazon Basics Dog and Puppy Pads, Leak-proof 5-Layer Pee Pads with Quick-dry Surface for Potty Training</p>
          <p class=\\"text-capitalize\\">pet clean-up and odor control</p>
          <p class=\\"font-weight-bold text-h6\\">24.82</p>
        </div>
        <div class=\\"slider-card\\">
          <div class=\\"v-responsive\\">
            <div class=\\"v-responsive__sizer\\"></div>
            <!---->
            <div class=\\"v-responsive__content\\">
              <div class=\\"v-responsive v-img v-img--booting\\" style=\\"height: 220px;\\">
                <div class=\\"v-responsive__sizer\\"></div>
                <transition-stub name=\\"fade-transition\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><img class=\\"v-img__img v-img__img--cover\\" src=\\"http://pet-shop.buckhill.com.hr/api/v1/file/1af8dd04-b8fe-3dad-8c5a-50e93f7ad1ec\\" style=\\"display: none;\\"></transition-stub>
                <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                  <!---->
                </transition-stub>
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <p class=\\"text-decoration-underline mt-4 mb-2 font-weight-bold text-h6\\">Fresh Wave Odor Removing Gel, 15 oz. (Pack of 2)</p>
          <p class=\\"text-capitalize\\">pet clean-up and odor control</p>
          <p class=\\"font-weight-bold text-h6\\">16.92</p>
        </div>
        <div class=\\"slider-card\\">
          <div class=\\"v-responsive\\">
            <div class=\\"v-responsive__sizer\\"></div>
            <!---->
            <div class=\\"v-responsive__content\\">
              <div class=\\"v-responsive v-img v-img--booting\\" style=\\"height: 220px;\\">
                <div class=\\"v-responsive__sizer\\"></div>
                <transition-stub name=\\"fade-transition\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><img class=\\"v-img__img v-img__img--cover\\" src=\\"http://pet-shop.buckhill.com.hr/api/v1/file/0fc80f84-36dd-3a6e-aacd-45d35438e7a0\\" style=\\"display: none;\\"></transition-stub>
                <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                  <!---->
                </transition-stub>
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <p class=\\"text-decoration-underline mt-4 mb-2 font-weight-bold text-h6\\">Nature’s Miracle Cage Cleaner 24 fl oz, Small Animal Formula, Cleans And Deodorizes Small Animal Cages, 2nd Edition</p>
          <p class=\\"text-capitalize\\">pet clean-up and odor control</p>
          <p class=\\"font-weight-bold text-h6\\">28.93</p>
        </div>
        <div class=\\"slider-card\\">
          <div class=\\"v-responsive\\">
            <div class=\\"v-responsive__sizer\\"></div>
            <!---->
            <div class=\\"v-responsive__content\\">
              <div class=\\"v-responsive v-img v-img--booting\\" style=\\"height: 220px;\\">
                <div class=\\"v-responsive__sizer\\"></div>
                <transition-stub name=\\"fade-transition\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><img class=\\"v-img__img v-img__img--cover\\" src=\\"http://pet-shop.buckhill.com.hr/api/v1/file/fd628a19-87ff-38a7-b665-20df1c3c5642\\" style=\\"display: none;\\"></transition-stub>
                <transition-stub name=\\"fade-transition\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
                  <!---->
                </transition-stub>
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
          <p class=\\"text-decoration-underline mt-4 mb-2 font-weight-bold text-h6\\">Black Diamond Stoneworks Ur-in Control Eliminates Urine Odors – Removes Cat, Dog, Pet, Odors Human Smells from Carpet</p>
          <p class=\\"text-capitalize\\">pet clean-up and odor control</p>
          <p class=\\"font-weight-bold text-h6\\">19.38</p>
        </div>
      </div>
    </div>
    <!---->
  </div>
</div>"
`;
